ui/src/components/OasModal/OasModal.tsx

```diff
@@ -0,0 +1,95 @@
+import { Box, Fade, FormControl, MenuItem, Modal } from "@material-ui/core";
+import { useEffect, useState } from "react";
+import { RedocStandalone } from "redoc";
+import Api from "../../helpers/api";
+import { Select } from "../UI/Select";
+import closeIcon from "../assets/closeIcon.svg";
+import { arrayElementsComparetion } from "../../helpers/utils";
+import { toast } from 'react-toastify';
+
+const api = new Api();
+
+const OasModal = ({ openModal, handleCloseModal, entries }) => {
+  const [oasServices, setOASservices] = useState([]);
+  const [selectedOASService, setSelectedOASService] = useState("");
+  const [oasService, setServiceOAS] = useState(null);
+
+  const noOasServiceSelectedMessage = "Please Select OasService";
+
+  useEffect(() => {
+    (async () => {
+      try {
+        const services = await api.getOASAServices();
+        if (!arrayElementsComparetion(oasServices, services)) setOASservices(services);
+      } catch (e) {
+        toast.error(e.message);
+        console.error(e);
+      }
+    })();
+  }, [entries,oasServices]);
```

Comment:
please re-think about the logic here:
1. it looks like you fetch the services on each new entry - that means a LOT of requests all the time
2. you can't put oasServices as dep where you use setOasServices

Reply:
1. I know its a lot of requests, the thing is, that I want to update the oasServices when there is new service. The only way that a new service is arrived is through new entries isn't it? maybe I can call the useEffect every time the "openModal" changes. It means that while someone in the modal the services wont updated and he will need to close and open again. what do you think? 
2. Its make sense but its weird, when I remove it I get the next warning in the console: 
"src/components/OasModal/OasModal.tsx
  Line 28:6:  React Hook useEffect has a missing dependency: 'oasServices'. Either include it or remove the dependency array  react-hooks/exhaustive-deps"

Reply:
1. that sounds like a good solution - go for it

Reply:
2. think about useCallback with the response of api.getOASAServices() as a dependency. that means you will have a useEffect that set a state holds the response of this API and you will update oasServices only via the useCallback

Reply:
What do you think about 2.?

Reply:
did you meant like this? 
  
const getOasServices = useCallback(async () => {
    try {
      const services = await api.getOasServices();
      if (!arrayElementsComparetion(oasServices, services)) setOASservices(services);
    } catch (e) {
      toast.error(e.message);
      console.error(e);
    }
  },[]);

  useEffect(() => {
      getOasServices();
  },[openModal]);

Reply:
I thought about putting this const services = await api.getOasServices();
inside the useEffect and then put the service response in a state and then pass this state as a dep for the useCallback.
but the solution you wrote may work as well - test it and if all looks good, we can go with ur solution